Error [ERR_MODULE_NOT_FOUND]: Cannot find module '.../prettier/internal/cli.mjs'

With recent Tag I see the following crash:
```
prettier.................................................................Failed
- hook id: prettier
- exit code: 1

node:internal/modules/esm/resolve:283
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/pc/clone/SEUvnMIkSDGFhKzXE86TPA/node_env-system/lib/node_modules/prettier/internal/cli.mjs' imported from /pc/clone/SEUvnMIkSDGFhKzXE86TPA/node_env-system/lib/node_modules/pre-commit-prettier/run-prettier.js
    at finalizeResolution (node:internal/modules/esm/resolve:283:11)
    at moduleResolve (node:internal/modules/esm/resolve:952:10)
    at defaultResolve (node:internal/modules/esm/resolve:1188:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:642:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:591:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:574:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:236:38)
    at ModuleJob._link (node:internal/modules/esm/module_job:130:49) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///pc/clone/SEUvnMIkSDGFhKzXE86TPA/node_env-system/lib/node_modules/prettier/internal/cli.mjs'
}

Node.js v20.19.2
```
see: https://results.pre-commit.ci/run/github/486408006/1751920579.dZUANJ6CRpqibtc7pdRQFA

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -51,7 +51,14 @@ jobs:
         shell: bash
 
       - name: Run pre-commit-mirror-maker
-        run: pre-commit-mirror . --language=node --package-name=prettier --types=text --id=prettier --entry "pre-commit-prettier --write --ignore-unknown"
+        run: |
+          pre-commit-mirror \
+            . \
+            --language=node \
+            --package-name=prettier \
+            --types=text \
+            --id=prettier \
+            --entry "pre-commit-prettier --write --ignore-unknown --experimental-cli --no-cache"
         shell: bash
 
       - name: Push to repository

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -6,6 +6,7 @@ repos:
     hooks:
       - id: prettier
         files: \.(json|xml|html|yaml|yml|js)$
+        exclude: ^\.pre-commit-hooks\.yaml$
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v5.0.0
     hooks:

--- a/.pre-commit-hooks.yaml
+++ b/.pre-commit-hooks.yaml
@@ -1,7 +1,7 @@
 -   id: prettier
     name: prettier
     description: ''
-    entry: pre-commit-prettier --write --ignore-unknown
+    entry: pre-commit-prettier --write --ignore-unknown --experimental-cli --no-cache
     language: node
     'types': [text]
     args: []

--- a/run-prettier.js
+++ b/run-prettier.js
@@ -79,13 +79,6 @@ const additionalArguments = plugins
     return ["--plugin", pluginImportFileUrl.toString()];
   });
 
-// Use '--experimental-cli' if it's available AND 'PRETTIER_EXPERIMENTAL_CLI' isn't set.
-// ('PRETTIER_EXPERIMENTAL_CLI' is a string, and "0" is truthy.)
-// Assume that, if '--experimental-cli' isn't available, it's now the default.
-if (!process.argv.includes("--experimental-cli") && !process.env.PRETTIER_EXPERIMENTAL_CLI) {
-  additionalArguments.push("--experimental-cli");
-}
-
 // the first two items of process.argv are reserved (the node.exe and the current file name) so insert after them
 process.argv.splice(2, 0, ...additionalArguments);
 

--- a/run-prettier.js
+++ b/run-prettier.js
@@ -4,7 +4,7 @@
 
 import { readFileSync, readdirSync, existsSync } from "node:fs";
 import path from "node:path";
-import { run as runPrettier } from "prettier/internal/cli.mjs";
+import { createRequire } from "node:module";
 
 const nodePath = path.resolve(process.env.NODE_PATH);
 const pluginsPath = path.resolve(nodePath, "@prettier");
@@ -90,4 +90,11 @@ process.stdout.write = function (message, ...optionalParams) {
   originalWrite.apply(process.stdout, [message, ...optionalParams]);
 };
 
-runPrettier();
+// Use '--experimental-cli' if it's available AND 'PRETTIER_EXPERIMENTAL_CLI' isn't set.
+// ('PRETTIER_EXPERIMENTAL_CLI' is a string, and "0" is truthy.)
+// Assume that, if '--experimental-cli' isn't available, it's now the default.
+if (!process.argv.includes("--experimental-cli") && !process.env.PRETTIER_EXPERIMENTAL_CLI) {
+  process.argv.push("--experimental-cli");
+}
+
+createRequire(import.meta.url)("prettier/bin/prettier.cjs");

--- a/run-prettier.js
+++ b/run-prettier.js
@@ -5,6 +5,7 @@
 import { readFileSync, readdirSync, existsSync } from "node:fs";
 import path from "node:path";
 import { createRequire } from "node:module";
+import url from "node:url";
 
 const nodePath = path.resolve(process.env.NODE_PATH);
 const pluginsPath = path.resolve(nodePath, "@prettier");
@@ -73,8 +74,17 @@ const additionalArguments = plugins
       console.error(`Error '${error}' reading or parsing package.json from '${pluginDirectoryPath}'`);
       pluginImportPath = pluginDirectoryPath;
     }
-    return ["--plugin", pluginImportPath];
+
+    const pluginImportFileUrl = url.pathToFileURL(pluginImportPath);
+    return ["--plugin", pluginImportFileUrl.toString()];
   });
+
+// Use '--experimental-cli' if it's available AND 'PRETTIER_EXPERIMENTAL_CLI' isn't set.
+// ('PRETTIER_EXPERIMENTAL_CLI' is a string, and "0" is truthy.)
+// Assume that, if '--experimental-cli' isn't available, it's now the default.
+if (!process.argv.includes("--experimental-cli") && !process.env.PRETTIER_EXPERIMENTAL_CLI) {
+  additionalArguments.push("--experimental-cli");
+}
 
 // the first two items of process.argv are reserved (the node.exe and the current file name) so insert after them
 process.argv.splice(2, 0, ...additionalArguments);
@@ -90,11 +100,6 @@ process.stdout.write = function (message, ...optionalParams) {
   originalWrite.apply(process.stdout, [message, ...optionalParams]);
 };
 
-// Use '--experimental-cli' if it's available AND 'PRETTIER_EXPERIMENTAL_CLI' isn't set.
-// ('PRETTIER_EXPERIMENTAL_CLI' is a string, and "0" is truthy.)
-// Assume that, if '--experimental-cli' isn't available, it's now the default.
-if (!process.argv.includes("--experimental-cli") && !process.env.PRETTIER_EXPERIMENTAL_CLI) {
-  process.argv.push("--experimental-cli");
-}
-
-createRequire(import.meta.url)("prettier/bin/prettier.cjs");
+const requireImport = createRequire(import.meta.url);
+const runPrettier = requireImport("prettier/bin/prettier.cjs");
+await runPrettier;

